pkg/microservice/aslan/core/service/service/helm.go

```diff
@@ -326,11 +328,132 @@ func CreateOrUpdateHelmService(projectName string, args *HelmServiceCreationArgs
 		return CreateOrUpdateHelmServiceFromGitRepo(projectName, args, logger)
 	case LoadFromChartTemplate:
 		return CreateOrUpdateHelmServiceFromChartTemplate(projectName, args, logger)
+	case LoadFromChartRepo:
+		return CreateOrUpdateHelmServiceFromChartRepo(projectName, args, logger)
 	default:
 		return nil, fmt.Errorf("invalid source")
 	}
 }
 
+func CreateOrUpdateHelmServiceFromChartRepo(projectName string, args *HelmServiceCreationArgs, log *zap.SugaredLogger) (*BulkHelmServiceCreationResponse, error) {
+	chartRepoArgs, ok := args.CreateFrom.(*CreateFromChartRepo)
+	if !ok {
+		return nil, e.ErrCreateTemplate.AddDesc("invalid argument")
+	}
+
+	chartRepo, err := commonrepo.NewHelmRepoColl().Find(&commonrepo.HelmRepoFindOption{RepoName: chartRepoArgs.ChartRepoName})
+	if err != nil {
+		log.Errorf("failed to query chart-repo info, productName: %s, err: %s", projectName, err)
+		return nil, e.ErrCreateTemplate.AddDesc(fmt.Sprintf("failed to query chart-repo info, productName: %s, repoName: %s", projectName, chartRepoArgs.ChartRepoName))
+	}
+
+	chartClient, err := helmclient.NewHelmChartRepoClient(chartRepo.URL, chartRepo.Username, chartRepo.Password)
+	if err != nil {
+		return nil, e.ErrCreateTemplate.AddErr(errors.Wrapf(err, "failed to init chart client for repo: %s", chartRepo.RepoName))
+	}
+
+	index, err := chartClient.FetchIndexYaml()
+	if err != nil {
+		return nil, e.ErrCreateTemplate.AddErr(err)
+	}
+
+	// validate chart with specific version exists in repo
+	foundChart := false
+	for name, entries := range index.Entries {
+		if name != chartRepoArgs.ChartName {
+			continue
+		}
+		for _, entry := range entries {
+			if entry.Version == chartRepoArgs.ChartVersion {
+				foundChart = true
+				break
+			}
+		}
+		break
```

Comment:
the data structure of index.Entries is a nested map like this: map[chartName][chartVersion], so we can break the loop since we have found the data with specific chartName